semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -196,4 +257,257 @@ public static boolean isValidUrl(String url) {
+    if (url.isPresent() && path.isEmpty()) {
+
+      try {
+        String pathval = getNormalizedUrl(url.get()).getPath();
+        if (StringUtils.isBlank(pathval)) {
+          pathval = SLASH;
+        }
+        return Optional.of(removeTrailingSlash(pathval));
+      } catch (MalformedURLException e) {
+        e.printStackTrace();
+      }
+    }
+    return path;
+  }
+
+  private static Optional<String> getHttpPathUtil(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String path : URL_PATH_ATTRIBUTES) {
+      if (attributeValueMap.get(path) != null) {
+        String s = attributeValueMap.get(path).getValue();
+        if (StringUtils.isNotBlank(s) && s.startsWith(SLASH)) {
+          return getPathFromUrlObject(s).map(str -> removeTrailingSlash(str));
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpMethod(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String method : METHOD_ATTRIBUTES) {
+      if (attributeValueMap.get(method) != null
+          && !StringUtils.isBlank(attributeValueMap.get(method).getValue())) {
+        return Optional.of(attributeValueMap.get(method).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpScheme(Event event) {
+
+    Optional<String> url = getHttpUrlUtil(event);
+    if (url.isPresent() && isAbsoluteUrl(url.get())) {
+      try {
+        return Optional.of(getNormalizedUrl(url.get()).getProtocol());
+      } catch (MalformedURLException e) {
+        e.printStackTrace();
+      }
+    }
+    return getHttpSchemeUtil(event);
+  }
+
+  private static Optional<String> getHttpSchemeUtil(Event event) {
+
+    if (event.getAttributes() == null || event.getAttributes().getAttributeMap() == null) {
+      return Optional.empty();
+    }
+
+    Map<String, AttributeValue> attributeValueMap = event.getAttributes().getAttributeMap();
+    for (String scheme : SCHEME_ATTRIBUTES) {
+      if (attributeValueMap.get(scheme) != null
+          && !StringUtils.isEmpty(attributeValueMap.get(scheme).getValue())) {
+        return Optional.of(attributeValueMap.get(scheme).getValue());
+      }
+    }
+    return Optional.empty();
+  }
+
+  public static Optional<String> getHttpUrl(Event event) {
+    Optional<String> url = getHttpUrlUtil(event);
+    if (url.isPresent() && !isAbsoluteUrl(url.get())) {
+      return Optional.empty();
+    }
+    return getHttpUrlUtil(event);
+  }
+
+  //  input url to populateurlparts
+  private static Optional<String> getHttpUrlUtil(Event event) {
```

Comment:
Can we have a better function name? May be `getNormalizedHttpUrl`? Or `getResolvedHttpUrl`?

Reply:
Renamed appropriately